File: frontend/src/stores/agent.ts
/** Agent 状态管理 */
import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
import { agentApi } from '@/api'
import type { AgentInfo } from '@/types'

export const useAgentStore = defineStore('agent', () => {
  const agents = ref<AgentInfo[]>([])
  const isLoading = ref(false)
  const error = ref<string | null>(null)

  async function loadAgents() {
    try {
      isLoading.value = true
      error.value = null
      agents.value = await agentApi.getAgents()
    } catch (err: any) {
      error.value = err.response?.data?.detail || '加载 Agent 列表失败'
      throw err
    } finally {
      isLoading.value = false
    }
  }

  function getAgentByMode(mode: string) {
    return agents.value.find(a => a.mode === mode)
  }

  return {
    agents,
    isLoading,
    error,
    loadAgents,
    getAgentByMode
  }
})

File: api/schemas/agent.py
"""Agent 相关的响应模型。"""

from typing import Optional

from pydantic import BaseModel


class AgentInfo(BaseModel):
    """单个 Agent 的信息。"""

    mode: str
    """Agent 模式标识，用于 API 调用"""

    name: str
    """Agent 显示名称"""

    description: Optional[str] = None
    """Agent 功能描述"""


class AgentList(BaseModel):
    """Agent 列表响应。"""

    agents: list[AgentInfo]
    """可用 Agent 列表"""
